### ISSUE TITLE
`hotUpdate` does not guard against adding new modules

### ISSUE THREAD

**Issue by HerringtonDarkholme** [vuejs/vuex#312]
I think this is an edge case, but it might worth reporting. 

Using vue@next and vuex@next

```javascript
var Vue = require('vue')
var Vuex = require('vuex')

Vue.use(Vuex)

var store = new Vuex.Store({})
store.hotUpdate({
  modules: {module: {}}
})

// TypeError: Cannot set property 'module' of undefined
```

**Comment by LinusBorg** [comment]
Hi @HerringtonDarkholme,

Thank your for filing this issue.

Can you elaborate a bit when/how this error happens? Did you realy inline the (vuex) module objects like you show, or is this supposed to be shortened pseudocode?

Hot module replacement is meant replace actual (commonJS) modules, not inline objects.

Also, your issue title refers to "new modules" while the code in the issue seems to be about submodules. Can you clarify?

Code for real reproduction would be helpful, too.

**Comment by HerringtonDarkholme** [comment]
Thank you for your quick response.

I just find it when trying to type check vuex's code. And compiler complains this error. I haven't see any real world case, since hot reloaded modules are usually installed on the root store.

**Comment by LinusBorg** [comment]
Thanks for replying, but that still does not make things clear for me:

* Does this only happen when trying to hot reload a module that has a submodule?
* Does this happen when I add a new module that has a submodule, or for any modules?
* ...?

**Comment by HerringtonDarkholme** [comment]
This happens when hot reloading a store that does not have any module, and the new option has a non-empty `modules` option.

I updated my contrived code example to avoid the word submodule. The problem is not relevant to submodule.

**Comment by LinusBorg** [comment]
Thanks!

**Comment by ktsn** [comment]
I think `hotUpdate` should only **update** modules and it is suitable to provide warning to encourage to reload a browser in this situation because it is difficult to detect removed modules and its state.

Imagine we will rename a module on development. If `hotUpdate` can add a new module, the renamed module is added as a new name but the old-name module is still exist with its state on the store. I think this may make confusion.

**Comment by ktsn** [comment]
Tha patch is merged (#316)
